lib/rules/quote-props.js

```diff
@@ -59,14 +71,62 @@ module.exports = function(context) {
         }
     }
 
-    return {
-        "Property": MODE === "as-needed" ? asNeeded : always
-    };
+    /**
+     * Ensures that an object's keys are consistenly quoted, optionally checks for redundancy of quotes
+     * @param   {ASTNode} node Property AST node
+     * @param   {boolean} checkQuotesRedundancy Whether to check quotes' redundancy
+     * @returns {void}
+     */
+    function consistent(node, checkQuotesRedundancy) {
+        var quotes = false,
+            lackOfQuotes = false,
+            necessaryQuotes = false;
+
+        node.properties.forEach(function(property) {
+            var key = property.key,
+                tokens;
+
+            if (key.type === "Literal" && typeof key.value === "string") {
+                quotes = true;
+                if (checkQuotesRedundancy) {
+                    try {
+                        tokens = espree.tokenize(key.value);
+                    } catch (e) {
+                        necessaryQuotes = true;
+                        return;
+                    }
+                    necessaryQuotes = necessaryQuotes || !areQuotesRedundant(tokens);
+                }
+            } else {
+                lackOfQuotes = true;
+            }
+
+            if (quotes && lackOfQuotes) {
+                context.report(node, "Inconsistently quoted property `{{key}}` found.", {
+                    key: key.name || key.value
+                });
+            }
+        });
+
+        if (checkQuotesRedundancy && quotes && !necessaryQuotes) {
+            context.report(node, "Properties shouldn't be quoted as all quotes are redundant.");
+        }
+    }
+
+    if (["always-if-needed", "consistent"].indexOf(MODE) >= 0) {
+        listeners.ObjectExpression = function(node) {
+            consistent(node, MODE === "always-if-needed");
+        };
+    } else {
+        listeners.Property = MODE === "as-needed" ? asNeeded : always;
```

Comment:
It's really hard to tell what nodes are being listened for in this format. Can you rework so it looks more like it did before?